src/core/request/deltafetch.js

```diff
@@ -46,132 +42,68 @@ export class DeltaFetchRequest extends KinveyRequest {
 
   execute() {
     return repositoryProvider.getOfflineRepository()
-      .then(repo => repo.read(this._getCollectionFromUrl(), this.query))
-      .catch((error) => {
-        if (!(error instanceof NotFoundError)) {
-          throw error;
+      .then((repo) => {
+        const collectionName = this._getCollectionFromUrl();
+        let queryCacheQuery = new Query()
```

Comment:
There are a couple of things I am not clear on here:

- tagged datastores - is it clear how they would work in this regard?
- the serialization of queries and general contents of the "query cache" collection

Reply:
I need to add a unit test for using a tagged datastore with delta fetch and make changes if things don't behave as expected. Good catch.